sdk/core/core/src/az_curl_adapter.c

```diff
@@ -296,6 +296,9 @@ AZ_NODISCARD az_result az_http_client_send_request_impl(
   if (az_succeeded(result)) {
     AZ_RETURN_IF_FAILED(az_span_builder_append(&response_builder, AZ_STR_ZERO));
   }
+  // update http_response value with the actual response size to avoid garbage
+  // https://github.com/Azure/azure-sdk-for-c/issues/219
+  response->value = az_span_builder_mut_result(&response_builder);
```

Comment:
The size is only going to decrease over time, as you reuse the response, and it may introduce new kind of bugs.
I think changing value to span_builder solves all the problems.

Reply:
we should never re-use span . we reuse the entire buffer for a new response.

we want to keep az_http_response struc that can eventually contain more than just one span to response. That's why we provide az_http_response_init so users don't reuse this structure but init one for every use

Reply:
When we retry an operation, we MUST clear out whatever is in the span so this MUST be possible and the retry policy code MUST execute the function to clear it.

Reply:
@JeffreyRichter , can we have Retry Policy to check if response is a 200 and then it takes care of resizing the span back to user?
I am thinking we can add a response_size field to http_response and transport policy would just write in it the size of the response. Then retry policy can use it to make it the size of the span.
I will update this PR with that approach so you can take comment on it

Reply:
The retry policy MUST ALWAYS reset the span before every request. It should NOT be looking for status codes (like 200). Let's just add a function to az_http_response called az_http_response_clear(...) that resets the size of the span back to 0 but leaves the capacity as is.

Reply:
How does retry policy knows when to retry request? 
I though it would check if response code is not 200, then it would re-send request.
The other option I can imagine is retrying only if transport policy returns something different than AZ_OK?

Reply:
The retry policy has a loop which executes once per try. Inside the loop it calls the next policy. Just before this call, it should reset the span. It can do this even on the 1st try.